### ISSUE TITLE
DNS Proxy disabled when using Standard Firewall SKU but not setting `address_management_prefix`

### ISSUE THREAD

**Issue by andrewdmay** [Azure/terraform-azurerm-caf-enterprise-scale#854]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 1.6.1

**azure provider**: 3.77.0

**module**: 5.0.0

### Description

When testing an upgrade to version `5.0.0` of the module the plan showed that it would disable the DNS proxy in the Firewall policy.

#### Describe the bug

PR #845 changed the logic around adding the Firewall management subnet to based upon `address_management_prefix` not being blank  instead of the SKU. As part of these changes it also changed the condition for the DNS proxy, but I think that should remain based upon the SKU as it's a feature available only for Standard and Premium SKUs.

Explicitly enabling DNS proxy by setting `enable_dns_proxy = true` in the Firewall config has no effect.

#### Steps to Reproduce

1. Deploy connectivity resources using version 4.2.0 of the module, enabling Azure Firewall and using the Standard SKU. Configure the `address_prefix` but not the `address_management_prefix`:
   ```
          azure_firewall = {
            enabled = true
            config = {
              address_prefix        = var.firewall_subnet_cidr
              sku_tier                 = "Standard"
              threat_intelligence_mode = "Alert"
            }
          }
   ```
2. Verify that DNS proxy is enabled by default
3. Upgrade module to 5.0.0, apply changes, DNS proxy will be disabled. Setting `enable_dns_proxy = true` does not re-enable it.

**Comment by hlokensgard** [comment]
Is there any workaround before a bugfix to keep the DNS proxy enabled? @matt-FFFFFF @andrewdmay ?

**Comment by matt-FFFFFF** [comment]
Released in 5.0.2
